src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -956,3 +943,87 @@ def assign_last_name_ids(self, pop: pd.DataFrame) -> pd.Series:
+        )
+        households_to_update_idx = population.index[
+            population["household_id"].isin(household_ids_without_reference_person)
+        ]
+
+        # Find oldest member in each household and make them new reference person
+        # This is a series with household_id as the index and the new reference person as the value
+        new_reference_persons = (
+            population.loc[households_to_update_idx].groupby(["household_id"])["age"].idxmax()
+        )
+        # Preserve old relationship of new reference person before assigning them as new reference persons
+        new_reference_person_prev_relationship = population.loc[
+            new_reference_persons, ["household_id", "relation_to_household_head"]
+        ]
+        population.loc[
+            new_reference_persons, "relation_to_household_head"
+        ] = "Reference person"
+
+        # Update simulants born in simulation as biological children to their mother if mother is new reference person
+        biological_children_idx = population.index[
+            (population["date_of_birth"] > self.start_time)
+            & (
+                population["guardian_1"]
+                == population["household_id"].map(
+                    new_reference_person_prev_relationship.reset_index().set_index(
+                        "household_id"
+                    )["index"]
+                )
+            )
+        ].intersection(households_to_update_idx)
+        population.loc[
+            biological_children_idx, "relation_to_household_head"
+        ] = "Biological child"
+
+        # Update other household members relationship to new reference person
+        for relationship in new_reference_person_prev_relationship[
+            "relation_to_household_head"
+        ].unique():
+            relationship_map = self.reference_person_update_relationships_map.loc[
+                self.reference_person_update_relationships_map[
+                    "new_reference_person_relationship_to_old_reference_person"
+                ]
+                == relationship
+            ].set_index("relationship_to_old_reference_person")
+
+            household_ids_to_update = new_reference_person_prev_relationship.loc[
+                new_reference_person_prev_relationship["relation_to_household_head"]
+                == relationship,
+                "household_id",
+            ]
+            simulants_to_update_idx = (
+                population.index[population["household_id"].isin(household_ids_to_update)]
+                .difference(new_reference_person_prev_relationship.index)
+                .difference(biological_children_idx)
+            )
+            # Update relationships
+            population.loc[
+                simulants_to_update_idx, "relation_to_household_head"
+            ] = population["relation_to_household_head"].map(
+                relationship_map["relationship_to_new_reference_person"]
+            )
```

Comment:
```suggestion
            ).fillna("Other nonrelative")
```

Reply:
I'm putting this outside of the for loop so it only has to search for null values and fill them once vs every time it goes through the loop.  I'm not sure if that will actually make it faster because in all of my testing I have yet to get a simulant where I would hit this but that is the idea.

Reply:
The operation is O(n), so it the only difference between doing it on each partition of `households_to_update_idx` and once on the whole thing would be any constant-time overhead of calling the `fillna` method. I'm guessing this is _very_ minimal in comparison to the operations done in this method. I think within the for loop is more readable, but it's just a style preference so nbd.